crates/hyperqueue/src/server/state.rs

```diff
@@ -129,22 +166,16 @@ impl State {
                 return None;
             }
         };
-        self.base_task_id_to_job_id.remove(&job.base_task_id);
+        for (_, base_task_id) in &job.submit_descs {
+            self.base_task_id_to_job_id.remove(base_task_id);
+        }
         Some(job)
     }
 
     pub fn get_job_mut_by_tako_task_id(&mut self, task_id: TakoTaskId) -> Option<&mut Job> {
         let job_id: JobId = *self.base_task_id_to_job_id.range(..=task_id).next_back()?.1;
         let job = self.jobs.get_mut(&job_id)?;
-        if task_id
-            < TakoTaskId::new(
-                job.base_task_id.as_num() + job.n_tasks() as <TaskId as ItemId>::IdType,
-            )
-        {
-            Some(job)
-        } else {
-            None
-        }
+        Some(job)
```

Comment:
Yes, it was removed because of multiple base_task_ids. It was just a sanity check to fail earlier, so it was not technically needed but was a kind of early assert.